src/Microsoft.TemplateEngine.Abstractions/TemplatePackage/IManagedTemplatePackage.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.TemplateEngine.Abstractions.Installer;
+
+namespace Microsoft.TemplateEngine.Abstractions.TemplatePackage
+{
+    /// <summary>
+    /// Represents the package that can be managed by <see cref="IManagedTemplatePackageProvider"/>. <see cref="IManagedTemplatePackageProvider"/> can additionally install, update and uninstall template package.
+    /// </summary>
+    public interface IManagedTemplatePackage : ITemplatePackage
+    {
+        /// <summary>
+        /// Gets the name to be used when displaying template package in UI.
+        /// </summary>
+        string DisplayName { get; }
+
+        /// <summary>
+        /// Gets the identifier of template package.
+        /// Identifier should be unique in scope of <see cref="IManagedTemplatePackageProvider"/> that manages the <see cref="IManagedTemplatePackage"/>.
+        /// </summary>
+        /// <remarks>
+        /// This can be NuGet PackageId, path to .nupkg, folder name, depends on <see cref="IInstaller"/> implementation.
+        /// </remarks>
+        string Identifier { get; }
+
+        /// <summary>
+        /// Gets <see cref="IInstaller"/> that installed the template package.
+        /// This serves as helper for grouping template package by installer so caller doesn't need to keep track of installer->template package relation.
+        /// </summary>
+        IInstaller Installer { get; }
+
+        /// <summary>
+        /// Gets <see cref="IManagedTemplatePackageProvider"/> that manages the template package.
+        /// This serves as helper for grouping packages by <see cref="IManagedTemplatePackageProvider"/>
+        /// so caller doesn't need to keep track of "managed provider"->"managed package" relation.
+        /// </summary>
+        IManagedTemplatePackageProvider ManagedProvider { get; }
```

Comment:
As description already explains it, its same as for Installer, its helper to make it easier to keep track of which provider/installer is responsible for this package, otherwise we would need to keep track which makes code complicated since https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.conditionalweaktable-2?view=net-5.0 would need to be used and bunch of code would look weird, hence, lets just have this 2 properties and move on was decision...

Reply:
I don't know how complicated the things get without this. But this feels like soon enough, we will look back and ask "why these fields are in this interface?". This is a public interface, so we won't get the chance to design this often. It sounds like a good idea to do it once and do it good.
I may be wrong with on this, so getting 3rd opinions could also help.

Reply:
I guess it comes down to, how hard it is to implement this for whoever is implementing this interface, and my feeling is, its super easy...